### PR TITLE
Add possibility to set an arbitrary acceptance radius 

### LINKED ISSUE #713: Add setting for mission item acceptance radius
It would be nice to set the acceptance radius for the mission items.

This cam up in: http://discuss.px4.io/t/mission-item-acceptance-radius/10218

### PR DESCRIPTION
**Pull request by ad-snow**
1. adds _acceptance_radius member variable to MissionItemImpl class
2. adds functions to set and get the acceptance radius
3. changes the get_mavlink_param2() function to return the acceptance radius if it is set,
   if it is not, it will return the predefined values (3 or 1 meter) depending on the _fly_through boolean, as it was before

Resolves #713.

### CODE REVIEW

File: plugins/mission/include/plugins/mission/mission_item.h [hunk 1]
```diff
@@ -52,6 +52,13 @@ class MissionItem {
      */
     void set_fly_through(bool fly_through);
 
+    /**
+     * @brief Set the acceptance radius property of a mission item.
+     *
+     * @param radius Radius around the mission_item where it will be considered as reached
+     */
+    void set_acceptance_radius(float radius);
```
Comment (JonasVautherin) [inline plugins/mission/include/plugins/mission/mission_item.h]: I would add the unit to be consistent with the others in the file (e.g. `altitude_m`).

```suggestion
    void set_acceptance_radius(float radius_m);
```

File: plugins/mission/include/plugins/mission/mission_item.h [hunk 2]
```diff
@@ -153,6 +160,13 @@ class MissionItem {
      */
     bool get_fly_through() const;
 
+    /**
+     * @brief Get the acceptance radius of a mission item.
+     *
+     * @return Acceptance radius in meters
+     */
+    float get_acceptance_radius() const;
```
Comment (JonasVautherin) [inline plugins/mission/include/plugins/mission/mission_item.h]: Same here

```suggestion
    float get_acceptance_radius_m() const;
```

File: plugins/mission/include/plugins/mission/mission_item.h [hunk 3]
```diff
@@ -153,6 +160,13 @@ class MissionItem {
      */
     bool get_fly_through() const;
 
+    /**
+     * @brief Get the acceptance radius of a mission item.
+     *
+     * @return Acceptance radius in meters
```
Comment (julianoes) [inline plugins/mission/include/plugins/mission/mission_item.h]: Nitpick: period at the end.

File: plugins/mission/include/plugins/mission/mission_item.h [hunk 4]
```diff
@@ -52,6 +52,13 @@ class MissionItem {
      */
     void set_fly_through(bool fly_through);
 
+    /**
+     * @brief Set the acceptance radius property of a mission item.
+     *
+     * @param radius Radius around the mission_item where it will be considered as reached
```
Comment (julianoes) [inline plugins/mission/include/plugins/mission/mission_item.h]: Nitpick: please specify unit and end sentence with period.

File: plugins/mission/mission_item_impl.h [hunk 5]
```diff
@@ -48,6 +50,7 @@ class MissionItemImpl {
     float _relative_altitude_m = NAN;
     float _speed_m_s = NAN;
     bool _fly_through = false;
+    float _acceptance_radius_m = 0.0;
```
Comment (julianoes) [inline plugins/mission/mission_item_impl.h]: I would like to have this consistent with the other properties. So I would vote to set the default to `NAN`.

File: plugins/mission/mission_item_impl.cpp [hunk 6]
```diff
@@ -85,9 +90,13 @@ float MissionItemImpl::get_mavlink_param1() const
 float MissionItemImpl::get_mavlink_param2() const
 {
     float acceptance_radius_m;
-    if (_fly_through) {
+    if (_acceptance_radius_m) {
```
Comment (julianoes) [inline plugins/mission/mission_item_impl.cpp]: I don't like to check a float for `== 0`, so I'd rather check for `std::isfinite()` and then use it if it's set.

**Comment by JonasVautherin** [comment]
Looking good @ad-snow. I believe the last thing is that the CI is failing because of the style. Could you try to run the following?

```
make fix_style
```

File: plugins/mission/include/plugins/mission/mission_item.h (hunk 3 above)
Comment (ad-snow) [reply plugins/mission/include/plugins/mission/mission_item.h]: done

File: plugins/mission/include/plugins/mission/mission_item.h (hunk 4 above)
Comment (ad-snow) [reply plugins/mission/include/plugins/mission/mission_item.h]: done

File: plugins/mission/mission_item_impl.h (hunk 5 above)
Comment (ad-snow) [reply plugins/mission/mission_item_impl.h]: makes sense

File: plugins/mission/mission_item_impl.cpp (hunk 6 above)
Comment (ad-snow) [reply plugins/mission/mission_item_impl.cpp]: got it

**Comment by julianoes** [comment]
Yep, only thing to fix is `make fix_style`:
```
+ make fix_style
Changed plugins/mission/include/plugins/mission/mission_item.h:
58c58,59
<      * @param radius_m Radius in meters around the mission_item where it will be considered as reached.
---
>      * @param radius_m Radius in meters around the mission_item where it will be considered as
>      * reached.
Makefile:140: recipe for target 'fix_style' failed
make: *** [fix_style] Error 1
```

**Comment by ad-snow** [comment]
Thanks for the support, I'm trying to get make fix_style running on my machine, I'll address the errors as soon as possible

**Comment by julianoes** [comment]
@ad-snow thanks a lot for the contribution again!

**Pull request merged**